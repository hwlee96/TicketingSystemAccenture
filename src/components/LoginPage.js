import React from "react";
import "../styles/LoginPage.css";
import aLogo from "../Resources/Icons/aLogo.svg";

import { Redirect } from "react-router-dom";

class LoginPage extends React.Component {
  constructor() {
    super();
    this.state = {
      username: "",
      password: "",
      error: false,
      redirect: false
    };
    this.handleChange = this.handleChange.bind(this);

    this.onSubmit = this.onSubmit.bind(this);
  }

  handleChange = e => {
    // {name,value} = e.target;
    this.setState({
      [e.target.name]: e.target.value
    });
  };

  onSubmit = e => {
    e.preventDefault();

    // {name,value} = e.target;
    if (
      (this.state.username === "benjamin" ||
        this.state.username === "joseph") &&
      this.state.password === "password"
    ) {
      //here i will change the isAuth state
      this.props.handleAuth();

      //redirect page
      this.setState({
        redirect: true
      });
    } else {
      this.setState({
        error: true
      });
    }
  };

  render() {
    if (this.state.redirect) {
      if (this.state.username === "joseph") {
        return <Redirect to="/crequests" />;
      }
      return <Redirect to="/requests" />;
    }

    let errorText;
    this.state.error
      ? (errorText = "Username or password is wrong, please try again")
      : (errorText = "");
    return (
      <div className="loginbody">
        <div className="login-page">
          <div className="form">
            <img
              src={aLogo}
              alt="accenture icon"
              className="accenture-logo-login"
            />
            <p className="accenture-text-login">Accenture Requests Assistant</p>

            <form className="register-form">
              <input type="text" placeholder="name" />
              <input type="password" placeholder="password" />
              <input type="text" placeholder="email address" />
              <button>create</button>
              <p className="message">
                Already registered? <a href="#">Sign In</a>
              </p>
            </form>
            <form className="login-form" onSubmit={this.onSubmit}>
              <input
                type="text"
                placeholder="Username"
                id="username-jest" //added for jest
                name="username"
                value={this.state.username}
                onChange={this.handleChange}
              />
              <input
                type="password"
                id="password-jest" //added for jest
                placeholder="Password"
                name="password"
                value={this.state.password}
                onChange={this.handleChange}
              />

              <p className="error-text">{errorText}</p>
              <button>login</button>
              <p className="message">
                Not registered? <a href="#">Create an account</a>
              </p>
            </form>
          </div>
        </div>
      </div>
    );
  }
}

export default LoginPage;
