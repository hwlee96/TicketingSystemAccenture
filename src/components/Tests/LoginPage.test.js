import React from "react";
import renderer from "react-test-renderer";
import { shallow, mount, render, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import LoginPage from "../LoginPage";

configure({ adapter: new Adapter() });

it("renders correctly", () => {
  const tree = renderer.create(<LoginPage />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders a usernameinput", () => {
  expect(shallow(<LoginPage />).find("#username-jest").length).toEqual(1);
});

it("renders a password input", () => {
  expect(shallow(<LoginPage />).find("#password-jest").length).toEqual(1);
});

describe("Username input test", () => {
  it("should respond to change event and change the state of the LoginPage Component", () => {
    const wrapper = shallow(<LoginPage />);
    wrapper.find("#username-jest").simulate("change", {
      target: { name: "username", value: "blah@gmail.com" }
    });
    expect(wrapper.state("username")).toEqual("blah@gmail.com");
  });
});

describe("Password input test", () => {
  it("should respond to change event and change the state of the LoginPage Component", () => {
    const wrapper = shallow(<LoginPage />);
    wrapper.find("#password-jest").simulate("change", {
      target: { name: "password", value: "blahblah" }
    });
    expect(wrapper.state("password")).toEqual("blahblah");
  });
});
