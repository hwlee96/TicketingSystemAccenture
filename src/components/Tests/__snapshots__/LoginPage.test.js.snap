// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="loginbody"
>
  <div
    className="login-page"
  >
    <div
      className="form"
    >
      <img
        alt="accenture icon"
        className="accenture-logo-login"
        src="aLogo.svg"
      />
      <p
        className="accenture-text-login"
      >
        Accenture Requests Assistant
      </p>
      <form
        className="register-form"
      >
        <input
          placeholder="name"
          type="text"
        />
        <input
          placeholder="password"
          type="password"
        />
        <input
          placeholder="email address"
          type="text"
        />
        <button>
          create
        </button>
        <p
          className="message"
        >
          Already registered? 
          <a
            href="#"
          >
            Sign In
          </a>
        </p>
      </form>
      <form
        className="login-form"
        onSubmit={[Function]}
      >
        <input
          id="username-jest"
          name="username"
          onChange={[Function]}
          placeholder="Username"
          type="text"
          value=""
        />
        <input
          id="password-jest"
          name="password"
          onChange={[Function]}
          placeholder="Password"
          type="password"
          value=""
        />
        <p
          className="error-text"
        >
          
        </p>
        <button>
          login
        </button>
        <p
          className="message"
        >
          Not registered? 
          <a
            href="#"
          >
            Create an account
          </a>
        </p>
      </form>
    </div>
  </div>
</div>
`;
